File: ios/Classes/ButtonClickHandler.swift
import Foundation
import Flutter

class ButtonClickHandler: NSObject, FlutterStreamHandler{
    var sink: FlutterEventSink?

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }

    @objc func buttonClicked(title: String?, type: String?, payload: String?) {
        guard let sink = sink else { return }
        let dictionary = [
            "TITLE": title,
            "TYPE": type,
            "PAYLOAD": payload
        ]
        sink(dictionary)
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/io/verloop/verloop_flutter_sdk/ButtonClickHandler.kt
package io.verloop.verloop_flutter_sdk

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.EventChannel


class ButtonClickHandler : EventChannel.StreamHandler {
    private val uiThreadHandler: Handler = Handler(Looper.getMainLooper())
    var sink: EventChannel.EventSink? = null

    fun buttonClicked(title: String?, type: String?, payload: String?) {
        uiThreadHandler.post {
            sink?.success(
                mapOf(
                    "TITLE" to title,
                    "TYPE" to type,
                    "PAYLOAD" to payload
                )
            )
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}